### ISSUE TITLE
how to get Server Push(not encraypted) at nghttp2.org by http://nghttp2.org/, not by https://nghttp2.org/

### ISSUE THREAD

**Issue by y123456yz** [nghttp2/nghttp2#849]
I want to get the server push frame, and I want to look the frame by wireshark using tcpdump. but if use https://nghttp2.org, wireshark can not decode because the package encrypted;

3q

**Comment by y123456yz** [comment]
@JimmyRittenborg  can you help me, thanks

**Comment by y123456yz** [comment]
[root@s10-2-x-4 nghttp2-master]# ./src/nghttp -vnu  http://nghttp2.org/index.html
[  0.073] Connected
[  0.073] HTTP Upgrade request
GET /index.html HTTP/1.1
host: nghttp2.org
connection: Upgrade, HTTP2-Settings
upgrade: h2c
http2-settings: AAMAAABkAAQAAP__
accept: */*
user-agent: nghttp2/1.21.0-DEV


[  0.147] HTTP Upgrade response
HTTP/1.1 101 Switching Protocols
Connection: Upgrade
Upgrade: h2c


[  0.147] HTTP Upgrade success
[  0.147] recv SETTINGS frame <length=18, flags=0x00, stream_id=0>
          (niv=3)
          [SETTINGS_MAX_CONCURRENT_STREAMS(0x03):100]
          [SETTINGS_INITIAL_WINDOW_SIZE(0x04):1048576]
          [SETTINGS_HEADER_TABLE_SIZE(0x01):8192]
[  0.147] send SETTINGS frame <length=12, flags=0x00, stream_id=0>
          (niv=2)
          [SETTINGS_MAX_CONCURRENT_STREAMS(0x03):100]
          [SETTINGS_INITIAL_WINDOW_SIZE(0x04):65535]
[  0.147] send SETTINGS frame <length=0, flags=0x01, stream_id=0>
          ; ACK
          (niv=0)
[  0.147] send PRIORITY frame <length=5, flags=0x00, stream_id=3>
          (dep_stream_id=0, weight=201, exclusive=0)
[  0.147] send PRIORITY frame <length=5, flags=0x00, stream_id=5>
          (dep_stream_id=0, weight=101, exclusive=0)
[  0.147] send PRIORITY frame <length=5, flags=0x00, stream_id=7>
          (dep_stream_id=0, weight=1, exclusive=0)
[  0.147] send PRIORITY frame <length=5, flags=0x00, stream_id=9>
          (dep_stream_id=7, weight=1, exclusive=0)
[  0.147] send PRIORITY frame <length=5, flags=0x00, stream_id=11>
          (dep_stream_id=3, weight=1, exclusive=0)
[  0.147] send PRIORITY frame <length=5, flags=0x00, stream_id=1>
          (dep_stream_id=11, weight=16, exclusive=0)
[  0.148] recv (stream_id=1) :status: 200
[  0.148] recv (stream_id=1) date: Mon, 20 Mar 2017 10:31:23 GMT
[  0.148] recv (stream_id=1) content-type: text/html
[  0.148] recv (stream_id=1) last-modified: Mon, 20 Mar 2017 08:26:48 GMT
[  0.148] recv (stream_id=1) etag: "58cf9248-19ff"
[  0.148] recv (stream_id=1) accept-ranges: bytes
[  0.148] recv (stream_id=1) content-length: 6655
[  0.148] recv (stream_id=1) x-backend-header-rtt: 0.001453
[  0.148] recv (stream_id=1) server: nghttpx
[  0.148] recv (stream_id=1) via: 2 nghttpx
[  0.148] recv (stream_id=1) x-frame-options: SAMEORIGIN
[  0.148] recv (stream_id=1) x-xss-protection: 1; mode=block
[  0.148] recv (stream_id=1) x-content-type-options: nosniff
[  0.148] recv HEADERS frame <length=198, flags=0x04, stream_id=1>
          ; END_HEADERS
          (padlen=0)
          ; First response header
[  0.218] recv DATA frame <length=6655, flags=0x01, stream_id=1>
          ; END_STREAM
[  0.218] recv SETTINGS frame <length=0, flags=0x01, stream_id=0>
          ; ACK
          (niv=0)
[  0.219] send GOAWAY frame <length=8, flags=0x00, stream_id=0>
          (last_stream_id=0, error_code=NO_ERROR(0x00), opaque_data(0)=[])
[root@s10-2-4-4 nghttp2-master]# 



I can not get server push frame

**Comment by tatsuhiro-t** [comment]
Try `nghttp -vn http://nghttp2.org/index.html`. Push is not triggered if HTTP Upgrade is used.

**Comment by y123456yz** [comment]
@tatsuhiro-t 
yes, I get the info, but not get server push frame. I want to get the server push frame that not encryped.
I want to look server push frame with wireshark.


if there have other method to get server push frame(not encryped),thanks

**Comment by tatsuhiro-t** [comment]
Accessing  http://nghttp2.org/index.html does not offer encryption.

**Comment by y123456yz** [comment]
@tatsuhiro-t   thanks, it is ok.
